frontend/server/src/main/java/org/pytorch/serve/wlm/WorkLoadManager.java

```diff
@@ -92,9 +94,67 @@ public int getNumRunningWorkers(ModelVersionName modelVersionName) {
             CompletableFuture<Integer> future = new CompletableFuture<>();
             int minWorker = model.getMinWorkers();
             int maxWorker = model.getMaxWorkers();
+            WorkerManagerThread workerManagerThread;
             List<WorkerThread> threads;
             if (minWorker == 0) {
+
+                logger.info("Min Worker is 0 - DHANAK");
+                workerManagerThread = workerManagers.remove(model.getModelVersionName());
                 threads = workers.remove(model.getModelVersionName());
+
+                if (workerManagerThread != null) {
+
+                    logger.info("Min Worker is not null - DHANAK");
+
+                    for (WorkerThread thread : threads) {
+                        workerManagerThread.scaleDown(thread.getLifeCycle().getPort());
```

Comment:
To my understanding, workerManagerThread.scaleDown is going to shut down the backend worker. It seems that the following code "workerkillprocess" does same thing.